### ISSUE TITLE
fetch-forms-from-google-drive may not complete its Promise cleanly

### ISSUE THREAD

**Issue by alxndrsn** [medic/medic-conf#88]
From @abbyad 

> are you using the fetch action in the same  command as the convert? I think it doesn't complete writing the xlsx file before trying to convert. I run them separately to avoid that issue.
>
> Eg `medic-conf fetch-forms-from-google-drive && medic-conf convert-app-forms`

**Comment by abbyad** [comment]
I am able to reproduce this error situation consistently:

```
$ medic-conf fetch-forms-from-google-drive convert-contact-forms
INFO Processing config in standard for undefined.
INFO Actions:
     - fetch-forms-from-google-drive
     - convert-contact-forms
INFO Extra args: undefined
INFO Current version: 1.17.14.  Checking NPM for updates…
WARN New version available!

          1.17.14 -> 1.18.0

     To install:

          npm install -g medic-conf

INFO Starting action: fetch-forms-from-google-drive…
INFO Exporting 1cq[redacted]SLs from google drive to forms/contact/PLACE_TYPE-create.xlsx…
INFO Successfully wrote forms/contact/PLACE_TYPE-create.xlsx.
INFO Exporting 1Vg[redacted]gpg from google drive to forms/contact/PLACE_TYPE-edit.xlsx…
INFO Successfully wrote forms/contact/PLACE_TYPE-edit.xlsx.
INFO fetch-forms-from-google-drive complete.
INFO Starting action: convert-contact-forms…
INFO Converting form ./forms/contact/clinic-create.xlsx …
Conversion complete!
INFO Converting form ./forms/contact/clinic-edit.xlsx …
Traceback (most recent call last):
  File "/usr/local/bin/xls2xform-medic", line 11, in <module>
    load_entry_point('pyxform-medic==0.9.24', 'console_scripts', 'xls2xform-medic')()
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2xform.py", line 88, in main_cli
    args.skip_validate)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2xform.py", line 18, in xls2xform_convert
    json_survey = xls2json.parse_file_to_json(xlsform_path, warnings=warnings)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2json.py", line 991, in parse_file_to_json
    workbook_dict = parse_file_to_workbook_dict(path, file_object)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2json.py", line 970, in parse_file_to_workbook_dict
    return xls_to_dict(file_object if file_object is not None else path)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2json_backends.py", line 41, in xls_to_dict
    workbook = xlrd.open_workbook(filename=path_or_file)
  File "/usr/local/lib/python2.7/dist-packages/xlrd/__init__.py", line 401, in open_workbook
    zf = zipfile.ZipFile(filename)
  File "/usr/lib/python2.7/zipfile.py", line 770, in __init__
    self._RealGetContents()
  File "/usr/lib/python2.7/zipfile.py", line 811, in _RealGetContents
    raise BadZipfile, "File is not a zip file"
zipfile.BadZipfile: File is not a zip file
ERROR Traceback (most recent call last):
  File "/usr/local/bin/xls2xform-medic", line 11, in <module>
    load_entry_point('pyxform-medic==0.9.24', 'console_scripts', 'xls2xform-medic')()
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2xform.py", line 88, in main_cli
    args.skip_validate)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2xform.py", line 18, in xls2xform_convert
    json_survey = xls2json.parse_file_to_json(xlsform_path, warnings=warnings)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2json.py", line 991, in parse_file_to_json
    workbook_dict = parse_file_to_workbook_dict(path, file_object)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2json.py", line 970, in parse_file_to_workbook_dict
    return xls_to_dict(file_object if file_object is not None else path)
  File "/usr/local/lib/python2.7/dist-packages/pyxform/xls2json_backends.py", line 41, in xls_to_dict
    workbook = xlrd.open_workbook(filename=path_or_file)
  File "/usr/local/lib/python2.7/dist-packages/xlrd/__init__.py", line 401, in open_workbook
    zf = zipfile.ZipFile(filename)
  File "/usr/lib/python2.7/zipfile.py", line 770, in __init__
    self._RealGetContents()
  File "/usr/lib/python2.7/zipfile.py", line 811, in _RealGetContents
    raise BadZipfile, "File is not a zip file"
zipfile.BadZipfile: File is not a zip file

```

Running as two separate commands  (eg `medic-conf fetch-forms-from-google-drive && medic-conf convert-contact-forms`) has yet to fail.

**Comment by alxndrsn** [comment]
I can't recreate this locally.  Please supply project configuration so I can confirm that we're testing with the same data.

**Comment by abbyad** [comment]
I was using the Standard config, from the [standard-for-2.16.x](https://github.com/medic/medic-webapp/tree/standard-for-2.16.x/config/standard) branch.

**Comment by alxndrsn** [comment]
I can recreate this.  Conversion fails on the final form that was exported from google drive.  I suspect the file stream isn't being closed properly, rather than an issue with promise completion.

**Comment by alxndrsn** [comment]
Can recreate with:
```
medic-conf fetch-forms-from-google-drive convert-contact-forms -- person-edit
```

**Comment by alxndrsn** [comment]
Should be fixed in `1.18.10`.
